### PR TITLE
Support no sharding config

### PR DESCRIPTION
**Pull request by zhaojuanmao**
supporting no sharding config to make it similar to DDP algorithm

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -103,25 +103,18 @@ class ShardingStrategy(Enum):
...
-                              parameters in GPU memory until backward computation is done. It
-                              inserts all_reduce after backward computation for synchronizing
-                              gradients. Sharded optimizer states are updated locally.
-    NO_SHARD(future support): This is similar to PyTorch `DistributedDataParallel` API.
-                              Parameters, gradients and optimizer states are replicated
-                              among ranks, all_reduce is inserted after backward computation
-                              is done for synchronizing gradients. Full optimizer states
-                              are updated in each rank.
+    NO_SHARD: This is similar to PyTorch `DistributedDataParallel` API. Parameters, gradients
+              and optimizer states are replicated among ranks, all_reduce is inserted after
```
Comment (mrshenli) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:107]: nit: same as on the previous PR. Double quote `DistributedDataParallel` and `all_reduce` to display as code.

**Review by mrshenli — APPROVED** [review]
Reviewed on #76628, stamp to unblock. 

Please remove changes in `ideep` submodule before landing.

**Comment by zhaojuanmao** [comment]
> Reviewed on #76628, stamp to unblock.
> 
> Please remove changes in `ideep` submodule before landing.

how to remove changes in `ideep` submodule..

**Comment by zhaojuanmao** [comment]
> > Reviewed on #76628, stamp to unblock.
> > Please remove changes in `ideep` submodule before landing.
> 
> how to remove changes in `ideep` submodule..

removed

**Comment by zhaojuanmao** [comment]
the commits messed up:(, closing this, will create a new PR

**Pull request closed without merging**